### ISSUE TITLE
[HOLD Apple SSO] [$1000] Desktop - infinite loading when opened from deeplink

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#19581]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Break down in numbered steps

1. Logout both web and desktop app
2. Enter https://staging.new.expensify.com/concierge in browser address bar
3. Open desktop app from popup

## Expected Result:

Desktop app redirects to login page

## Actual Result:

App keeps loading forever

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [X] MacOS / Desktop

**Version Number:** v1.3.17-1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/36083550/4d100df9-8f51-44d9-a4f5-4b4332de07bb

**Expensify/Expensify Issue URL:**
**Issue reported by:** @0xmiroslav 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1684924597264749

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~013f0e5cffe3d877da</li>
        <li>Upwork Job ID: 1661639425574531072</li>
        <li>Last Price Increase: 2023-05-25</li>
    </ul>
</details>

**Comment by koko57** [comment]
Hi I'm Agata from Callstack - expert contributor group - maybe I could help with this one?

**Comment by mountiny** [comment]
@koko57 great I think we should put this on hold for the navigation refactor then.

**Comment by mountiny** [comment]
Making this weekly, aiming to merge the refactor at the end of the week

**Comment by mountiny** [comment]
SWM guys we ooo and we were holding off for the RenderATL so unfortunately no. They are back on track so aiming to merge it Tuesday or Wednesday and let the avalanche go

**Comment by koko57** [comment]
Hi @mountiny! I've noticed that https://github.com/Expensify/App/pull/19263 is already merged.I'll be OOO until the end of this week, but I'll check into this issue when I return on Monday.

**Comment by mountiny** [comment]
Lovely, thanks!

**Comment by koko57** [comment]
@mountiny this one is unfortunately still reproducible :/ I'll continue my investigation here

**Comment by mountiny** [comment]
Making this back daily

**Comment by koko57** [comment]
I think I've found the root cause, I will post a proposal soon 🙂

**Comment by koko57** [comment]
I've found a potential fix for this issue, will post the proposal tomorrow 🙂

**Comment by koko57** [comment]
@mountiny just to make sure - we want the popup to appear even the user is not logged in (before logging in)?

**Comment by koko57** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

When opening the desktop app from the popup the app instead redirecting to Sign In page shows an infinite loader. It happens when the app is opened from the ‘/concierge’ link.

### What is the root cause of that problem?

Every time we open the app from the popup a new [expensifyDeepLinkUrl](https://github.com/Expensify/App/blob/23978e4142bcaed94162e94159c61d946f3467dd/src/components/DeeplinkWrapper/index.website.js#L60) is created pointing to the ‘/transition’ route. It also includes a param ‘exitTo’ which is created from the route that the app was opened in the browser - in this case ‘concierge’.

Because we’re logged off from the web app we don’t have shortLivedAuthToken this [condition](https://github.com/Expensify/App/blob/23978e4142bcaed94162e94159c61d946f3467dd/src/pages/LogInWithShortLivedAuthTokenPage.js#L52-L55) is skipped. As we have the ‘exitTo’ param the Navigation tries to navigate to the Concierge screen. 
The [PublicScreen’s](https://github.com/Expensify/App/blob/862955c2f15bce3c68d87c549b7af0e27a0eb526/src/libs/Navigation/AppNavigator/PublicScreens.js#L12) navigation stack doesn’t have this route so it cannot navigate to the Concierge page. The FullScreenLoadingIndicator is displayed because we’re passing isTokenValid: true as defaultProps, so it matches this [condition](https://github.com/Expensify/App/blob/862955c2f15bce3c68d87c549b7af0e27a0eb526/src/pages/LogInWithShortLivedAuthTokenPage.js#L66-L68) (in Onyx there’s no ‘isTokenValid’ record at that time)

It happens for all the routes available for the AuthScreens like ‘/settings’, ‘/r/{id}’ etc

### What changes do you think we should make in order to solve the problem?

I think we should check if the screen(route) is available in the current (PublicScreens) navigator. If so the user should be navigated to the given page, otherwise the default (home) page for each navigator should be displayed.
Changes that are necessary to apply this solution:
- Create an object mapping values for PublicScreens: ScreenName -> Route (path)
In this [condition](https://github.com/Expensify/App/blob/862955c2f15bce3c68d87c549b7af0e27a0eb526/src/pages/LogInWithShortLivedAuthTokenPage.js#L66-L68) 
- check if the mapping include the screen name corresponding to the path we want to navigate to
- If not - redirect to HomePage (here SignIn)

### What alternative solutions did you explore? (Optional)
I’ve tried to set defaultProp for isTokenValid -> a screen with an information that token is expired and a link to SignIn page is displayed

**Comment by koko57** [comment]
> I think your proposal looks good but just to confirm, after the redirect to the signin page and later successfull sign in, would the user land on the deeplinked page?

@mountiny no, sorry I didn't thought about that, but I can try to apply this flow tomorrow

> just to make sure - we want the popup to appear even the user is not logged in (before logging in)?

@0xmiroslav @mountiny I asked about this, because if the popup would appear only for the logged in user, no changes would be necessary here - opening electron app would work properly

**Comment by koko57** [comment]
Yeah, I think it can be on hold again, as after the apple sign-in PR the pop-up will show only after logging in on the web. The desktop app opens correctly when we're logged in on the web (we get a shortlived Auth token), so there's no need to change any code, the issue won't be occurring.

**Comment by mountiny** [comment]
Thanks, updated.

**Comment by mountiny** [comment]
Still on hold

**Comment by mountiny** [comment]
Still on hold

**Comment by laurenreidexpensify** [comment]
@mountiny we still on hold?

**Comment by mountiny** [comment]
Still on hold, the apple sso is not implemented yet

**Comment by mountiny** [comment]
correct its not there yet

**Comment by laurenreidexpensify** [comment]
Apple SSO is on Staging for web right?

**Comment by mountiny** [comment]
Correct i think @koko57 is ooo now, she can pick this up once back hopefully

**Comment by koko57** [comment]
@mountiny @laurenreidexpensify the issue is not reproducible now. So as long as we don't open the pop-up for unauthorized users the problem will not occur.
But as @0xmiroslav mentioned:
> Eventually, we will enable popup on unauthorized screens but that is a follow-up.
If we're going to do that, I guess we'll have to hold this issue again for this follow-up.

**Comment by mountiny** [comment]
Thanks for noting this is not reproducible. I think we can close this issue and when there will be some follow up it will be addressed there.

@0xmiroslav can you also double check if this is not repro anymore please?
